packages/cvxpy-base/patches/0001-Do-not-build-C-extensions.patch

```diff
@@ -0,0 +1,47 @@
+From 5029d6677b6790efe1b8362b32db3806e9272a7f Mon Sep 17 00:00:00 2001
+From: Philipp Schiele <44360364+phschiele@users.noreply.github.com>
+Date: Fri, 1 Mar 2024 22:51:50 +0000
+Subject: [PATCH] Do not build C++ extensions, make SciPy backend default
+
+---
+ cvxpy/settings.py | 2 +-
+ setup.py          | 3 +--
+ 2 files changed, 2 insertions(+), 3 deletions(-)
+
+diff --git a/cvxpy/settings.py b/cvxpy/settings.py
+index 1d1ef3eb1..662b227dc 100644
+--- a/cvxpy/settings.py
++++ b/cvxpy/settings.py
+@@ -182,7 +182,7 @@ SCIPY_CANON_BACKEND = "SCIPY"
+ RUST_CANON_BACKEND = "RUST"
+ CPP_CANON_BACKEND = "CPP"
+ 
+-DEFAULT_CANON_BACKEND = CPP_CANON_BACKEND
++DEFAULT_CANON_BACKEND = SCIPY_CANON_BACKEND
+ 
+ # Numerical tolerances
+ EIGVAL_TOL = 1e-10
+diff --git a/setup.py b/setup.py
+index 992ea6eb9..063fab949 100644
+--- a/setup.py
++++ b/setup.py
+@@ -5,7 +5,6 @@ import platform
+ import sys
+ import sysconfig
+ 
+-import setup.extensions as setup_extensions
+ import setup.versioning as setup_versioning
+ 
+ # BEFORE importing setuptools, remove MANIFEST. Otherwise it may not be
+@@ -56,7 +55,7 @@ VERSION = setup_versioning.VERSION
+ setup(
+     version=str(VERSION),
+     cmdclass={'build_ext': build_ext_cvxpy},
+-    ext_modules=[setup_extensions.cvxcore, setup_extensions.sparsecholesky],
++    ext_modules=[],
```

Comment:
Rather than keeping this patch in the Pyodide repository, it would be nice if you could add it to the cvxpy repository itself.

You may use `"PYODIDE" in os.environ` to detect if you are in Pyodide or building for Pyodide.

Reply:
Thanks a lot!

Opened a PR:
https://github.com/cvxpy/cvxpy/pull/2382/files

The other patched change is not part of the installation process, so it would require the `os.environ` lookup whenever the module is loaded. Is there another way to make such changes as part of the Pyodide build process?

Is it possible to point to a certain commit in our main branch instead of the latest release? The changes that are required to build for Pyodide are not released yet.

Reply:
> it would require the os.environ lookup whenever the module is loaded. Is there another way to make such changes as part of the Pyodide build process?

I am not sure if there is a way to do that in the build process. But I don't think it's too heavy to look up a dictionary once when the module is loaded.

Note that `"PYODIDE" in os.environ` works in the build time but not in runtime. In runtime, you need to do `sys.platform == "emscripten"` or `"pyodide" in sys.modules` instead (see: https://pyodide.org/en/stable/usage/faq.html#how-to-detect-that-code-is-run-with-pyodide)

Reply:
> Is it possible to point to a certain commit in our main branch instead of the latest release? The changes that are required to build for Pyodide are not released yet.

GitHub supports downloading an archive from a specific commit: https://stackoverflow.com/questions/13636559/how-to-download-zip-from-github-for-a-particular-commit-sha

Reply:
The CVXPY PR has been merged with your proposed changes.
Downloading from a commit that includes these changes for now: [e57847b](https://github.com/pyodide/pyodide/pull/4587/commits/e57847b44a90d708303b59c333880997eedee6ef)
We can switch to the PyPI version as soon as the next CVXPY release is out, which is scheduled for the end of March.
CC: @PTNobel